demos/digest/Makefile

```diff
@@ -0,0 +1,21 @@
+# Quick instruction:
+# To build against an OpenSSL built in the source tree, do this:
+#
+#    make OPENSSL_INCS_LOCATION=-I../../include OPENSSL_LIBS_LOCATION=-L../..
```

Comment:
Maybe use something less mouthful so it is easier to type. For example OSSL_INCLUDEDIR and OSSL_LIBDIR ? Or even just INCLUDEDIR LIBDIR?
Or you can simplify this even more by just putting -lssl and -lcrypto to a separate variable OSSL_LIBS and keep LDFLAGS and CFLAGS as the way how to set the -I and -L options.
BTW, `-lssl` is not really needed here.
`make CFLAGS=-I../../include LDFLAGS=-L../..`

Reply:
I agree with making this easier.  The change would need to propagate through to the other makefiles.

With the directory naming you're using (which I think is better), a `git mv demos/evp demos/cipher` is in order.